bundles/net.tourbook/src/net/tourbook/ui/views/tourDataEditor/TourDataEditorView.java

```diff
@@ -6786,6 +6830,45 @@ private void enableControls() {
       timeSliceTable.setEnabled(isDeviceTour);
    }
 
+   private void fillAirQualityCombo() {
+
+      for (int index = 0; index < IWeather.airQualityTexts.length; index++) {
+
+         final TableItem tableItem = new TableItem(_tableComboWeather_AirQuality.getTable(), SWT.READ_ONLY);
+
+         // set the column text
+         tableItem.setText(IWeather.airQualityTexts[index]);
+
+         int color;
+         switch (index) {
+         //Good
+         case 1:
+            color = SWT.COLOR_DARK_GREEN;
+            break;
+         //Fair
+         case 2:
+            color = SWT.COLOR_YELLOW;
```

Comment:
Fixed the text colors for the dark theme:
![image](https://user-images.githubusercontent.com/25235665/229327997-47930209-d10d-46ca-8f85-28091b825868.png)

However, on Linux, I don't have the same issue when the "Not defined" air quality is selected

![image](https://user-images.githubusercontent.com/25235665/229328015-46595f29-2d46-492f-b8a1-f6f61f11672b.png)

Reply:
> However, on Linux, I don't have the same issue when the "Not defined" air quality is selected
> 
> ![image](https://user-images.githubusercontent.com/25235665/229328015-46595f29-2d46-492f-b8a1-f6f61f11672b.png)

Sometimes when the focus is lost, then the dark themed colors are reverted, this is all over in MT

Reply:
For a dark background color, a bright foreground color should be used and vice versa

![image](https://user-images.githubusercontent.com/1283445/229332650-241d4dc4-0398-4e40-9697-3c0c7816ce89.png)

When the focus is lost, the the default colors are also displayed in W10

![image](https://user-images.githubusercontent.com/1283445/229332694-fcae41c1-b6d3-4e61-b675-db880d909356.png)

Reply:
https://user-images.githubusercontent.com/1283445/229332986-9cf655ce-aff1-4e0a-ac8c-2c5a0095c4cd.mp4

Reply:
> Sometimes when the focus is lost, then the dark themed colors are reverted (not painted), this is all over in MT

I wonder if that is related to this [known issue](https://www.eclipse.org/lists/nebula-dev/msg03559.html)

Reply:
> For a dark background color, a bright foreground color should be used and vice versa
> 

Except that I am encountering issues when the item in the `TableCombo` is hovered, then the text is not readable because it seems that a hovered item in the `TableCombo` is highlighted in blue automatically in any of the themes:

Example with the first item 'Fait'

Not hovered, foreground => It's readable
![image](https://user-images.githubusercontent.com/25235665/229644139-425a5f3f-fcd4-4a37-8c3a-2e5cf37f5ba3.png)



Hovered, foreground => It's not readable
![image](https://user-images.githubusercontent.com/25235665/229644170-e893defe-132a-46cd-b875-468b28df17f6.png)

Reply:
> Except that I am encountering issues when the item in the `TableCombo` is hovered, then the text is not readable because it seems that a hovered item in the `TableCombo` is highlighted in blue automatically in any of the themes:

I've seen this issue already in other places, it's not good but to solve it will propably be a nightmare

Can you exchange the `Poor` and `Very poor` background color because I think black should be the worst

Reply:
> > Sometimes when the focus is lost, then the dark themed colors are reverted (not painted), this is all over in MT
> 
> I wonder if that is related to this [known issue](https://www.eclipse.org/lists/nebula-dev/msg03559.html)

The css theming has still many issues, some are sometimes fixed, some are new regressions, e.g. https://github.com/eclipse-platform/eclipse.platform.ui/issues/340

Reply:
> Can you exchange the `Poor` and `Very poor` background color because I think black should be the worst

I don't understand, `Poor` is Red and `Very Poor` is grey. You want to exchange them so that `Very Poor` is Red ?

![image](https://user-images.githubusercontent.com/25235665/229821139-02de0ddf-af61-4256-a040-51e80dac9bfc.png)

![image](https://user-images.githubusercontent.com/25235665/229821576-7649628d-b3e0-4afe-9f33-a65c583130d3.png)

Reply:
`Poor` is not red when hovering, I missed this 😄 

As you can see also in your screenshot `<Not defined>` is not readable

![image](https://user-images.githubusercontent.com/1283445/229826219-d4231b40-4429-45ad-bdbf-171ae9a94505.png)

Reply:
> `Poor` is not red when hovering, I missed this 😄 -> this color is OK

That's what I was trying to say yesterday in my comment that you deleted, when hovering an item, then it's not always readable anymore:

![image](https://user-images.githubusercontent.com/25235665/229827457-1b5934d5-f895-4ba6-abe4-987cfb275b84.png)



> As you can see also in your screenshot `<Not defined>` is not readable
> 

If you don't hover over it, it seems readable to me
![image](https://user-images.githubusercontent.com/25235665/229827739-2da12760-464b-4b31-a3b0-a3522755d985.png)